### ISSUE TITLE
backend error w/class with constructor that doesn't properly satisfy interface

### ISSUE THREAD

**Issue by jvasileff** [ceylon/ceylon-compiler#2292]
In the ide, there is a redundant backend error for:

```ceylon
shared
interface I {
    shared formal void foo();
}

shared
class C satisfies I {
    shared new () { }
}
```

Ceylon Error: 
```
formal member 'foo' of 'I' not implemented in class hierarchy
```
but also, Ceylon Backend Error:
```
formal member 'foo' of 'I' not implemented in class hierarchy
```

**Comment by jvasileff** [comment]
Oh, it looks like the errors are reported for different lines (backend error line 6, tc error line 7), and the backend error disappears if the `shared` is not followed by '\n':

```ceylon
shared class C satisfies I {
    shared new () { }
}
```

**Comment by tombentley** [comment]
I'm not sure what's going on here, but I don't think there's a bug in the compiler.

1. javac doesn't use the "formal member" terminology, so that error cannot be from javac.
2. I cannot reproduce this behaviour in the CLI compiler.
3. The CLI compiler error recovery code actually implements a `foo()` method, so there really is nothing for javac to complain about.

I suspect this is actually a bug the IDE.

**Comment by FroMage** [comment]
Well, we should file it for the IDE then, no?

**Comment by tombentley** [comment]
Yes, sorry.
